src/Microsoft.AspNetCore.DataProtection.Extensions/DataProtectionProvider.cs

```diff
@@ -40,21 +49,25 @@ public static IDataProtectionProvider Create(DirectoryInfo keyDirectory)
             {
                 throw new ArgumentNullException(nameof(keyDirectory));
             }
-
             if (setupAction == null)
             {
                 throw new ArgumentNullException(nameof(setupAction));
             }
 
+            return CreateProvider(keyDirectory, setupAction);
+        }
+
+        private static IDataProtectionProvider CreateProvider(
+            DirectoryInfo keyDirectory,
+            Action<IDataProtectionBuilder> setupAction)
+        {
             // build the service collection
             var serviceCollection = new ServiceCollection();
             var builder = serviceCollection.AddDataProtection();
+
             builder.PersistKeysToFileSystem(keyDirectory);
```

Comment:
Heeeeeh! `PersistKeysToFileSystem` throws an exception when using a null value: https://github.com/aspnet/DataProtection/blob/dev/src/Microsoft.AspNetCore.DataProtection/DataProtectionBuilderExtensions.cs#L180.

Does the new overload even works?